src/ui/map.js

```diff
@@ -1045,18 +1056,52 @@ class Map extends Camera {
      */
     setProjection(projection?: ?ProjectionSpecification | string) {
         this._lazyInitEmptyStyle();
-        if (typeof projection === 'string') {
+        if (projection === undefined) { projection = null; } else if (typeof projection === 'string') {
```

Comment:
I don't have anything against one liner for a single `if` statement, but let's expand this one into separate 
statements.

Although, it's probably not really a hot debugging codepath here, it can help the debugging experience to be able to put breakpoints in separate lines based on the condition.
```suggestion
        if (projection === undefined) {
            projection = null;
        } else if (typeof projection === 'string') {
```